### ISSUE TITLE
InlinePanel reference docs are overly complicated

### ISSUE THREAD

**Issue by gasman** [wagtail/wagtail#9887]
### Pertinent section of the Wagtail docs

https://docs.wagtail.org/en/latest/reference/pages/panels.html#inline-panels-and-model-clusters

### Details

(Anyone picking this issue up should familiarise themselves with the [four modes of documentation](https://docs.wagtail.org/en/latest/contributing/documentation_modes.html).)

In the code example, the `RelatedLink` abstract model is unnecessary - the definitions in there could be incorporated directly into `BookPageRelatedLinks` instead. Defining an abstract model _might_ be good practice if you intend to put a "related links" section on several page types and want to avoid duplicating code (although it's arguably even better to just define the relation once against the base `wagtailcore.Page` model) - but the reader doesn't really need to know that at the point that they're just learning about `InlinePanel`. This leads to an overly long and complicated explanation of the code - it seems that the original author was aware of this, hence the ["we'll skip over this for now"](https://docs.wagtail.org/en/latest/reference/pages/panels.html#inlinepanel) text further up.

The deeper problem is that this documentation can't decide whether it wants to be 'how-to' or reference documentation. There's an existing (but fairly short) section about InlinePanel in the proper Usage Guide how-to section, at https://docs.wagtail.org/en/latest/topics/pages.html#inline-models - any explanations aimed at teaching the concepts should go there instead.

I suggest the following changes:

* Decide if there's anything worthwhile in https://docs.wagtail.org/en/latest/reference/pages/panels.html#inline-panels-and-model-clusters that isn't already covered by [the Usage Guide page](https://docs.wagtail.org/en/latest/topics/pages.html#inline-models), besides the abstract inheritance stuff - if so, incorporate that information into the "Inline models" section on the Usage Guide page
* Add a new section to [the Usage Guide page](https://docs.wagtail.org/en/latest/topics/pages.html#inline-models) underneath "Inline models", showing how the `BlogPageRelatedLink` model introduced there can be refactored into an abstract model. (This can borrow heavily from the abstract model example on https://docs.wagtail.org/en/latest/reference/pages/panels.html#inline-panels-and-model-clusters.) This could also mention the alternative approach, of defining the relation on the base `wagtailcore.Page` model so that it's available on all page types.
* Remove the now obsolete section at https://docs.wagtail.org/en/latest/reference/pages/panels.html#inline-panels-and-model-clusters
* Change https://docs.wagtail.org/en/latest/reference/pages/panels.html#inlinepanel to a brief description of the parameters available on `InlinePanel` - like we do for the other panel types - and include a link to the Usage Guide section for people who want a fuller explanation.

**Comment by vallabhtiwari** [comment]
Hi @gasman, I would like to work on this please...?

**Comment by gasman** [comment]
@vallabhtiwari Sure, go for it!

**Comment by 0saurabh0** [comment]
> This could also mention the alternative approach, of defining the relation on the base `wagtailcore.Page` model so that it's available on all page types.

@gasman can you explain it a bit more; an overview would be fine too.

**Comment by gasman** [comment]
@0saurabh0 Sure, happy to explain (but please give @vallabhtiwari a few more days before jumping in on this)...

Taking the example at https://docs.wagtail.org/en/latest/topics/pages.html#inline-models as a starting point, if we instead wrote:

```python
from django.db import models
from modelcluster.fields import ParentalKey
from wagtail.models import Orderable


class PageRelatedLink(Orderable):
    page = ParentalKey("wagtailcore.Page", on_delete=models.CASCADE, related_name='related_links')
    name = models.CharField(max_length=255)
    url = models.URLField()

    panels = [
        FieldPanel('name'),
        FieldPanel('url'),
    ]
```

(note the change of ParentalKey to "wagtailcore.Page") then the `related_links` relation will be set up on the base Page model, which means it will be available on all page types. Most likely we don't want to use it on _all_ page types (for example, related links are not very useful on a homepage) but it will only actually show up on the edit form for page types that have the `InlinePanel('related_links', label="Related links")` definition on them. So, for the page types where we do want related links, we can go ahead and add that line to `content_panels` - for all other page types, the relation will remain empty (because we're not providing editors with any way to fill it in) and will have no other effect on those pages.

**Comment by vallabhtiwari** [comment]
Hi @gasman , I think the only additional information given by the [#inline-panels-and-model-clusters](https://docs.wagtail.org/en/latest/reference/pages/panels.html#inline-panels-and-model-clusters) is about the parameters that ```InlinePanel```  accepts (except the abstract model stuff) and also, is very complicated as compared to [#inline-models](https://docs.wagtail.org/en/latest/topics/pages.html#inline-models) section. This could be incorporated in [#inlinepanel](https://docs.wagtail.org/en/latest/reference/pages/panels.html#inlinepanel) like you said and a link can be provided. And about the abstract model stuff, what heading should I give to the new section that needs to be underneath the "inline models" section.

**Comment by ArijeetBanerjee** [comment]
i want to work on this issue. kindly give me access to work on it.

**Comment by gasman** [comment]
@ArijeetBanerjee Please stop. Claiming 15+ issues (without checking which ones are already being worked on) is excessive.

**Comment by ArijeetBanerjee** [comment]
@gasman sorry my bad

**Comment by DipenSolanki01** [comment]
> Hi @gasman , I think the only additional information given by the [#inline-panels-and-model-clusters](https://docs.wagtail.org/en/latest/reference/pages/panels.html#inline-panels-and-model-clusters) is about the parameters that `InlinePanel` accepts (except the abstract model stuff) and also, is very complicated as compared to [#inline-models](https://docs.wagtail.org/en/latest/topics/pages.html#inline-models) section. This could be incorporated in [#inlinepanel](https://docs.wagtail.org/en/latest/reference/pages/panels.html#inlinepanel) like you said and a link can be provided. And about the abstract model stuff, what heading should I give to the new section that needs to be underneath the "inline models" section.

hi vallabh 
i have seen your work by finding you unexpectedly , u seem really good guy i am new around here and in my collage there are not any enthusiastic students like you about open source so can please so me around this or if possible become mentor or friend ,it will be good experience for u also and helpful for me ,u can message me on linkedin if you can spare some time that i have provided in my bio
thank you

**Comment by vallabhtiwari** [comment]
#9940

**Comment by MashiatK** [comment]
I want to work on this. If the issue is open, please assign it to me.

**Comment by gasman** [comment]
@MashiatK Please read the discussion above - there is already a pull request for this.

**Comment by gasman** [comment]
Fixed in #9940
